codeflash/optimization/function_optimizer.py

```diff
@@ -581,25 +592,61 @@ def write_code_and_helpers(original_code: str, original_helper_code: dict[Path,
                 f.write(original_helper_code[module_abspath])
 
     def reformat_code_and_helpers(
-        self, helper_functions: list[FunctionSource], path: Path, original_code: str
+        self,
+        preexisting_functions_by_filepath: dict[Path, set[tuple[str, tuple[FunctionParent,...]]]],
+        helper_functions: list[FunctionSource],
+        fto_path: Path,
+        original_code: str,
     ) -> tuple[str, dict[Path, str]]:
         should_sort_imports = not self.args.disable_imports_sorting
         if should_sort_imports and isort.code(original_code) != original_code:
             should_sort_imports = False
+        
+        paths = [fto_path] + list({hf.file_path for hf in helper_functions})
+        new_target_code = None
+        new_helper_code: dict[Path, str] = {}
+        for i, path in enumerate(paths):
+            unformatted_code = path.read_text(encoding="utf8")
+            code_ranges_unformatted = get_modification_code_ranges(
+                unformatted_code,
+                self.function_to_optimize,
+                preexisting_functions_by_filepath[path],
+                helper_functions,
+            )
+            formatted_code = format_code(self.args.formatter_cmds, path)
+            # Note: We do not need to refresh the code_context because we only use it to refer to names of original
+            # functions (even before optimization was applied) and filepaths, none of which is changing.
+            code_ranges_formatted = get_modification_code_ranges(
+                formatted_code,
+                self.function_to_optimize,
+                preexisting_functions_by_filepath[path],
+                helper_functions,
+            )
 
-        new_code = format_code(self.args.formatter_cmds, path)
-        if should_sort_imports:
-            new_code = sort_imports(new_code)
+            if len(code_ranges_formatted) != len(code_ranges_unformatted):
```

Comment:
Q: Can you help me understand the case on Why these have to be equal always?

Reply:
Just a defensive measure. The function to optimize and helpers are non-overlapping blocks of code, and should not become overlapping due to formatting. If this has happened, something has gone wrong in the formatting step or in the extraction step (which identifies function to optimize and helpers).